### PR TITLE
Add custom error handling, 404s and 500s

### PR DESCRIPTION
**Pull request by stormytuna**

### CODE REVIEW

**Comment by LSB616** [comment]
Will we need to handle the mongodb error codes?  

https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml

File: __tests__/app.test.js [hunk 1]
```diff
@@ -18,6 +18,18 @@ afterAll(() => {
   return disconnect();
 });
 
+describe("404s on non-existent endpoints", () => {
+  test("status:404, responds with an appropriate error message", () => {
```
Comment (blessinguche) [inline __tests__/app.test.js:22]: more tests for diifferent errors?

File: controllers/controllers.errors.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+exports.handleCustomErrors = (err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(error);
+  }
+};
+
+exports.handle404s = (req, res, next) => {
+  res.status(404).send({ msg: "Content not found" });
```
Comment (blessinguche) [inline controllers/controllers.errors.js:15]: needs a next to 500 and if statement to determine if is a 404

**Review by blessinguche — COMMENTED** [review]
overall good , need more secure error handling... look at error code (console.log())

File: controllers/controllers.errors.js (hunk 2 above)
Comment (stormytuna) [reply controllers/controllers.errors.js:15]: Shouldn't do, we only 404 on endpoints we dont recognise

File: __tests__/app.test.js (hunk 1 above)
Comment (stormytuna) [reply __tests__/app.test.js:22]: Error tests will be described in each endpoint, the only standalone test we need is 404 for now

**Comment by stormytuna** [comment]
We will need to test for mongodb errors, but I'm not sure which ones exactly, add them as and when needed

**Pull request merged**